### DISCUSSION TITLE
Return and display result from GET REST API

### DISCUSSION THREAD

**Discussion by NiviAppno** [Budibase/budibase#4133] [Help]
seemingly simple thing to do I would have thought

I have a REST API data source defined
I have a GET query with a variable as "binding" : https://myrestapi.com/{{variable}}
The above is working as expected.

In design, I have a simple form for the variable, and an action to execute the query
This, as well, is working.

All I want now is to display the response
I wasn't able to do this with a data provider "component"

**Comment by SantoshSrinivas79** [comment]
The data provider is the way to go. Did you set the `Data` source of the data provider to the REST query directly?

**Reply by mjashanks** [reply]
Agreed - this is the way to do it. There are some scenarios where you want to display the result of a `POST` request. You can do this, using the `Update State` action...

https://user-images.githubusercontent.com/3524181/150514913-2050447b-589b-4add-b6dc-b3996e043de5.mp4

**Reply by Jefflai0315** [reply]
@mjashanks could you show an example of how do you retrieve the id value "159571835"? 
so instead of showing the whole query result, I just want a single value.

**Reply by mjashanks** [reply]
So - following on from my video above - in the `Update State` action, you would use the binding `{{ Action 1.Query result.id }}`

Or `{{ Action 1.Query result.0.id }}` if your API is returning an array (this will choose the first item in the array).

**Comment by Jefflai0315** [comment]
![image](https://user-images.githubusercontent.com/79297365/154079272-f5909e83-1458-42bc-9403-d68fd1b6a232.png)
![image](https://user-images.githubusercontent.com/79297365/154079378-9866f35a-7d5f-4fa9-b530-070e569456d5.png)

I tried {{ Action 1.Query result.0.result }} and {{ Action 1.Query result.result }} , but doesnt work
